lib/GLPI/Agent/SNMP/MibSupport/LinuxAppliance.pm

```diff
@@ -257,6 +268,52 @@ sub run {
 
     my $firmware;
     if ($manufacturer eq 'Synology') {
+        
+        my $diskModels = $self->walk(dsmDiskModel)
+        or return;
+
+        my $diskNames = $self->walk(dsmDiskName)
+        or return;
+
+        my $volumesNames = $self->walk(dsmRaidName)
+        or return;
+
+        my $volumesFreeSizes = $self->walk(dsmRaidFreeSize)
+        or return;
+
+        my $volumesTotalSizes = $self->walk(dsmRaidTotalSize)
+        or return;
+
+        foreach my $key (keys(%{$diskModels})){
+
+            my $storage = {
+            TYPE    => 'disk',
+            };
+
+            my $name = trimWhitespace(getCanonicalString($diskNames->{$key}));
+            my $model = $diskModels->{$key};
+            my $manufacturer = getCanonicalManufacturer($model);
+            $storage->{MANUFACTURER} = $manufacturer; 
+            $storage->{MODEL} = $model . " - " . $name;
+            push @{$device->{STORAGES}}, $storage
+            if $storage->{MODEL};
```

Comment:
You have a wrong indentation on l.299, but this is indeed a wrong test. The test should be done on `$name` & `$model` to go on next key if nothing is found.

But can you explain why you want to add `$name` in `MODEL` ? Or can you give an example of a result ? It would even be nice to share a [snmpwalk output](https://glpi-agent.readthedocs.io/en/latest/bug-reporting.html#snmpwalk-output) so I can validate the PR.

```suggestion
            my $name = trimWhitespace(getCanonicalString($diskNames->{$key}))
                or next;
            my $model = $diskModels->{$key}
                or next;
            my $manufacturer = getCanonicalManufacturer($model);
            $storage->{MANUFACTURER} = $manufacturer; 
            $storage->{MODEL} = $model . " - " . $name;
            push @{$device->{STORAGES}}, $storage;
```